clients/src/test/java/org/apache/kafka/clients/consumer/internals/ShareConsumerImplTest.java

```diff
@@ -197,23 +212,34 @@ public void testWakeupBeforeCallingPoll() {
 
     @Test
     public void testWakeupAfterEmptyFetch() {
-        consumer = newConsumer();
+        SubscriptionState subscriptions = new SubscriptionState(new LogContext(), OffsetResetStrategy.NONE);
+        consumer = newConsumer(
+            mock(ShareFetchBuffer.class),
+            subscriptions,
+            "group-id",
+            "client-id");
         final String topicName = "foo";
-        final int partition = 3;
         doAnswer(invocation -> {
             consumer.wakeup();
             return ShareFetch.empty();
         }).doAnswer(invocation -> ShareFetch.empty()).when(fetchCollector).collect(any(ShareFetchBuffer.class));
 
-        consumer.subscribe(singletonList(topicName));
+        final List<String> subscriptionTopic = Collections.singletonList(topicName);
+        completeShareSubscriptionChangeApplicationEventSuccessfully(subscriptions, subscriptionTopic);
+        consumer.subscribe(subscriptionTopic);
 
         assertThrows(WakeupException.class, () -> consumer.poll(Duration.ofMinutes(1)));
         assertDoesNotThrow(() -> consumer.poll(Duration.ZERO));
     }
 
     @Test
     public void testWakeupAfterNonEmptyFetch() {
-        consumer = newConsumer();
+        SubscriptionState subscriptions = new SubscriptionState(new LogContext(), OffsetResetStrategy.NONE);
+        consumer = newConsumer(
```

Comment:
nit: a new helper might make sense since it looks like the same args in all these tests.